tests/hardhat/Admin/VBNBAdmin.ts

```diff
@@ -0,0 +1,124 @@
+import { FakeContract, MockContract, smock } from "@defi-wonderland/smock";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { BigNumber, Signer } from "ethers";
+
+import {
+  ComptrollerHarness,
+  ComptrollerHarness__factory,
+  IAccessControlManager,
+  IProtocolShareReserve,
+  InterestRateModelHarness,
+  MockVBNB,
+  MockVBNB__factory,
+  VBNBAdmin,
+  WBNB,
+  WBNB__factory,
+} from "../../../typechain";
+
+export const bigNumber18 = BigNumber.from("1000000000000000000"); // 1e18
+
+type SetupMarketFixture = {
+  comptroller: MockContract<ComptrollerHarness>;
+  vBNB: MockContract<MockVBNB>;
+  protocolShareReserve: FakeContract<IProtocolShareReserve>;
+  WBNB: MockContract<WBNB>;
+  VBNBAdmin: VBNBAdmin;
+  VBNBAdminAsVBNB: MockContract<MockVBNB>;
+};
+
+const setupMarketFixture = async (): Promise<SetupMarketFixture> => {
+  const [admin] = await ethers.getSigners();
+
+  const ComptrollerFactory = await smock.mock<ComptrollerHarness__factory>("ComptrollerHarness");
+  const comptroller = await ComptrollerFactory.deploy();
+
+  const interestRateModelHarnessFactory = await ethers.getContractFactory("InterestRateModelHarness");
+  const InterestRateModelHarness = (await interestRateModelHarnessFactory.deploy(
+    BigNumber.from(18).mul(5),
+  )) as InterestRateModelHarness;
+
+  const MockVBNBFactory = await smock.mock<MockVBNB__factory>("MockVBNB");
+
+  const mockVBNB = await MockVBNBFactory.deploy(
+    comptroller.address,
+    InterestRateModelHarness.address,
+    bigNumber18,
+    "Venus BNB",
+    "vBNB",
+    18,
+    admin.address,
+  );
+
+  const protocolShareReserve = await smock.fake<IProtocolShareReserve>("IProtocolShareReserve");
+
+  const WBNBFactory = await smock.mock<WBNB__factory>("WBNB");
+  const WBNB = await WBNBFactory.deploy();
+
+  const accessControl = await smock.fake<IAccessControlManager>("IAccessControlManager");
+  accessControl.isAllowedToCall.returns(true);
+
+  const VBNBAdminFactory = await ethers.getContractFactory("VBNBAdmin");
+  const VBNBAdmin: VBNBAdmin = await upgrades.deployProxy(VBNBAdminFactory, [
```

Comment:
It is suggested to test this against the live vBNB and network.
I guess this does not work because:
1. `transfer` has 2300 gas limit. VBNBAdmin might not be able to receive BNB
2. vBNB is not upgradable.

https://github.com/VenusProtocol/venus-protocol/blob/develop/contracts/Tokens/VTokens/VBNB.sol.archive#L147

Reply:
I did came across this issue before(`TransparentUpgradeableProxy`, that's the default).

Here is a discussion(from other users).
https://forum.openzeppelin.com/t/transparentupgradeableproxy-is-there-any-risk-in-omitting-admin-check-from-receive-function/27315